enhancements/hypershift/disaster-recovery/hcp-etcd-backup-api.md

```diff
@@ -0,0 +1,641 @@
+	  StoreEndpoint(*hyperv1.storeEndpoint) --> ServiceConfigType
+	  StoreEndpoint(*hyperv1.storeEndpoint) --> S3ServiceConfig("ServiceConfig --S3ServiceConfig--")
+	  StoreEndpoint(*hyperv1.storeEndpoint) --> AzureServiceConfig("ServiceConfig --AzureServiceConfig--")
+	  StoreEndpoint(*hyperv1.storeEndpoint) --> ODFServiceConfig("ServiceConfig --ODFServiceConfig--")
+	  StoreEndpoint(*hyperv1.storeEndpoint) --> Credentials("Credentials --SecretRef--")
+
+    ServiceConfigType --> S3ConfigType("S3ServiceConfig")
+    ServiceConfigType --> AzureServiceConfigType("AzureServiceConfig")
+    ServiceConfigType --> ODFServiceConfigType("ODFServiceConfig")
+
+    S3ServiceConfig --> ServiceURL("ServiceURL --string--")
+    S3ServiceConfig --> Region("Region --string--")
+    S3ServiceConfig --> Tags("Tags --slice--")
+
+    AzureServiceConfig --> AzFieldX(FieldX)
+    AzureServiceConfig --> AzFieldY(FieldY)
+    AzureServiceConfig --> AzFieldZ(FieldZ)
+
+    ODFServiceConfig --> ODFFieldX(FieldX)
+    ODFServiceConfig --> ODFFieldY(FieldY)
+    ODFServiceConfig --> ODFFieldZ(FieldZ)
+
+  end
+end
+```
+
+
+
+### Topology Considerations
+
+#### Hypershift / Hosted Control Planes
+
+This is implemented directly for Hypershift.
+
+#### Standalone Clusters
+
+N/A
+
+#### Single-node Deployments or MicroShift
+
+**SNO:** The ETCD backup API and Controllers will work in the same way having 1-N nodes.
+**Microshift:** N/A
+
+### Implementation Details/Notes/Constraints
+
+In this section we will dissect the behaviour of the different controller to be implemented.
+
+#### ETCDBackupController
+
+- The `ETCDBackupController` initiates with a series of validations:
+  - Checks if the backup has already been performed by examining the HC and HCP labels (further details provided at the end of the process).
+  - Validates access permissions to the PVC and available free space. Errors in either could result in log entries, a new status.condition for the HC and HCP, and an event.
+  - Verifies the existence and health status of the HostedCluster necessary for performing the backup, as well as the presence of the PVC.
+  - The controller creates a new PVC per backup and reschedules the backup for the next iteration, awaiting the status of Bound. This PVC will have a label to be identified as the new vessel for the backup (this label will be deleted later). (1 PVC per backup)
+    - If the PVC binding fails after a certain period, the controller will abandon the backup attempt and mark it as a failure.
+    - If the PVC it's bound to a PV and contains the label/annotation for "new backup", the controller will trigger the backup process.
+    - If the PVC it's bound to a PV and contains the label/annotation for "hash validation", the controller will perform a process to validate the existant backup.
+    - TBD: valorate to include a retryPolicy
+- The controller initiates the backup process by generating the ETCD backup object in the PVC specified in the manifest.
```

Comment:
I'd like to see a little more information about what we're expecting this controller to do - are we doing the backup in the controller process?

Reply:
Hm, are we saying that this controller creates the `config.openshift.io/v1alpha1:Backup` resource? If so, that resource already has the schedule built in. Why are we needing to do timing checks in our controller? Please clarify the separation of concerns here.

Reply:
This process generates the backup binary object that will be uploaded to the proper destination. The inner process is not 100% clear to me but basically it uses the `etcdctl` to create a backup.